plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -783,7 +783,27 @@ public void createSchema(ConnectorSession session, String schemaName, Map<String
     @Override
     public void dropSchema(ConnectorSession session, String schemaName)
```

Comment:
The more I think about it, the less sure I am that this is the right place to delete the files.

I think it might be better to include a `deleteData` parameter for `HiveMetastore.dropDatabase`, and pass true when there are no external files. That argument is already present on `HiveMetastore.dropTable` and `dropPartition`, too, so it wouldn't be entirely out of place.

I guess if we delete the file here, though, `HiveMetastore.dropDatabase` should note that it's not supposed to delete non-metadata files.